### PR TITLE
Update workshop materials

### LINKED ISSUE #796: update workshop materials
--------
Issue Description
--------
##### What did you expect to see happen?
The workshop should be up to date.

EDIT @alfoa : All the inputs in the workshop material should be part of our regression test system.

##### What did you see instead?
It is quite old now

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
close #796

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: doc/workshop/ExternalModels/projectile.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+# Simulates time-dependent track of a projectile through the air from start to 0,
+#     assuming no air resistance.
+#     Inputs:
+#       (x0,y0) - initial position
+#       v0 - initial total velocity
+#       ang - angle of initial motion, in degrees, with respect to flat ground
+#     Outputs:
+#       (x,y) - vector positions of projectile in time
+#       t - corresponding time steps
+#
```
Comment (PaulTalbot-INL) [inline doc/workshop/ExternalModels/projectile.py]: Can this file be merged with the one in AnalyticModels?

Comment (alfoa) [reply doc/workshop/ExternalModels/projectile.py]: agree

File: doc/workshop/ExternalModels/projectile.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    Calculates the analytic range.
+    @ In, v, float, velocity of the projectile
+    @ In, th, float, angle to the ground for initial projectile motion
+    @ In, y0, float, optional, initial height of projectile
+    @ In, g, float, optional, gravitational constant (m/s/s)
+    @ Out, prange, float, range
+  """
+  return v*np.cos(th)/g * (v*np.sin(th) + np.sqrt(v*v*np.sin(th)**2+2.*g*y0))
+
+def time_to_ground(v,th,y0=0,g=9.8):
```
Comment (alfoa) [inline doc/workshop/ExternalModels/projectile.py]: camelBack.... the new NQ1 standard will require that any source code we generate must follow RAVEN standards

File: doc/workshop/ExternalModels/projectile.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    Calculates the analytic time of flight
+    @ In, v, float, velocity of the projectile
+    @ In, th, float, angle to the ground for initial projectile motion
+    @ In, y0, float, optional, initial height of projectile
+    @ In, g, float, optional, gravitational constant (m/s/s)
+    @ Out, time_to_ground, float, time projectile is above the ground
+  """
+  return v*np.sin(th)/g + np.sqrt(v*v*np.sin(th)**2+2.*g*y0)/g
+
+def x_pos(x0,v,t):
```
Comment (alfoa) [inline doc/workshop/ExternalModels/projectile.py]: same as above

File: doc/workshop/ExternalModels/projectile.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  """
+    Calculates the x position in time
+    @ In, x0, float, initial horizontal position
+    @ In, v, float, velocity of the projectile
+    @ In, t, float, time of flight
+    @ Out, x_pos, float, horizontal position
+  """
+  return x0 + v*t
+
+def y_pos(y0,v,t):
```
Comment (alfoa) [inline doc/workshop/ExternalModels/projectile.py]: same as above

File: doc/workshop/advancedReliability/exercises/Inputs/1_adaptive_sampling.xml [hunk 5]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>Results</WorkingDir>
+    <Sequence>adaptive_sampling,plot</Sequence>
+    <batchSize>10</batchSize>
+  </RunInfo>
```
Comment (alfoa) [inline doc/workshop/advancedReliability/exercises/Inputs/1_adaptive_sampling.xml]: add the <TestInfo> block

File: doc/workshop/advancedReliability/exercises/Inputs/2_ls_probability_computation.xml [hunk 6]
```diff
@@ -0,0 +1,112 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>Results</WorkingDir>
+    <Sequence>load_ls,integral,plot</Sequence>
+    <batchSize>10</batchSize>
+  </RunInfo>
```
Comment (alfoa) [inline doc/workshop/advancedReliability/exercises/Inputs/2_ls_probability_computation.xml]: same as above (This comment is valid for all the input files)

File: doc/workshop/advancedReliability/inputs/tests [hunk 7]
```diff
@@ -0,0 +1,19 @@
+[Tests]
```
Comment (alfoa) [inline doc/workshop/advancedReliability/inputs/tests]: Do we need to push the database above (```doc/workshop/advancedReliability/inputs/resultsPB/out_db_adaptive.h5```)?

File: doc/workshop/codeCoupling/exercises/sampleProjectile/input.i [hunk 8]
```diff
@@ -0,0 +1,4 @@
+x0 = 0
```
Comment (alfoa) [inline doc/workshop/codeCoupling/exercises/sampleProjectile/input.i]: remove file ```doc/workshop/codeCoupling/exercises/sampleProjectile/.DS_Store``` above

File: doc/workshop/codeCoupling/exercises/sample_new_code_interface.xml [hunk 9]
```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>sampleProjectile</WorkingDir>
+    <Sequence>generate_data,plot</Sequence>
+    <batchSize>4</batchSize>
+  </RunInfo>
```
Comment (alfoa) [inline doc/workshop/codeCoupling/exercises/sample_new_code_interface.xml]: ```<TestInfo>```

File: doc/workshop/codeCoupling/projectile_model/projectile.py [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+    Calculates the analytic range.
+    @ In, v, float, velocity of the projectile
+    @ In, th, float, angle to the ground for initial projectile motion
+    @ In, y0, float, optional, initial height of projectile
+    @ In, g, float, optional, gravitational constant (m/s/s)
+    @ Out, prange, float, range
+  """
+  return v*np.cos(th)/g * (v*np.sin(th) + np.sqrt(v*v*np.sin(th)**2+2.*g*y0))
+
+def time_to_ground(v,th,y0=0,g=9.8):
```
Comment (alfoa) [inline doc/workshop/codeCoupling/projectile_model/projectile.py]: camelback...

Can we merge this model with the others?

File: doc/workshop/ensembleModels/inputs/ensembleExample/1_ensemble_model.xml [hunk 11]
```diff
@@ -0,0 +1,132 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Simulation verbosity="debug">
+
+    <RunInfo>
+
+	    <WorkingDir>results</WorkingDir>
```
Comment (alfoa) [inline doc/workshop/ensembleModels/inputs/ensembleExample/1_ensemble_model.xml]: indentation...

PS. ```<TestInfo>```

File: doc/workshop/ensembleModels/inputs/hybridExample/test_hybrid_model.xml [hunk 12]
```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="debug">
+  <RunInfo>
+    <WorkingDir>results</WorkingDir>
+    <Sequence>hybridRun,print</Sequence>
+    <batchSize>1</batchSize>
+    <internalParallel>False</internalParallel>
+  </RunInfo>
+
```
Comment (alfoa) [inline doc/workshop/ensembleModels/inputs/hybridExample/test_hybrid_model.xml]: ```<TestInfo>```

File: doc/workshop/forwardSampling/Inputs/1_sample_and_plot.xml [hunk 13]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>r1</WorkingDir>
+    <Sequence>generate_data,plot</Sequence>
+    <batchSize>4</batchSize>
+  </RunInfo>
+
```
Comment (alfoa) [inline doc/workshop/forwardSampling/Inputs/1_sample_and_plot.xml]: ```<TestInfo>```

File: doc/workshop/forwardSampling/Inputs/2_normal_distribution.xml [hunk 14]
```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>r2</WorkingDir>
+    <Sequence>generate_data,plot</Sequence>
+    <batchSize>4</batchSize>
+  </RunInfo>
+
```
Comment (alfoa) [inline doc/workshop/forwardSampling/Inputs/2_normal_distribution.xml]: ```<TestInfo>```

File: doc/workshop/forwardSampling/Inputs/3_initial_height.xml [hunk 15]
```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>r3</WorkingDir>
+    <Sequence>generate_data,plot</Sequence>
+    <batchSize>4</batchSize>
+  </RunInfo>
+
```
Comment (alfoa) [inline doc/workshop/forwardSampling/Inputs/3_initial_height.xml]: ```<TestInfo>```

File: doc/workshop/forwardSampling/Inputs/4_grid_sampler.xml [hunk 16]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>r4</WorkingDir>
+    <Sequence>generate_data,plot</Sequence>
+    <batchSize>4</batchSize>
+  </RunInfo>
+
```
Comment (alfoa) [inline doc/workshop/forwardSampling/Inputs/4_grid_sampler.xml]: ```<TestInfo>```

File: doc/workshop/forwardSampling/Inputs/5_basic_stats.xml [hunk 17]
```diff
@@ -0,0 +1,130 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <RunInfo>
+    <WorkingDir>r5</WorkingDir>
+    <Sequence>generate_data,stats,plot</Sequence>
+    <batchSize>4</batchSize>
+  </RunInfo>
+
```
Comment (alfoa) [inline doc/workshop/forwardSampling/Inputs/5_basic_stats.xml]: ```<TestInfo>```

File: doc/workshop/optimizer/exercise/plotWalk.py [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+          ix = line.index('angle')
+          iy = line.index('v0')
+          continue
+        data[case]['angle'].append(float(line[ix]))
+        data[case]['v0'].append(float(line[iy]))
+  except IOError:
+    cases = cases[:c]
+    break
+
+print data
```
Comment (alfoa) [inline doc/workshop/optimizer/exercise/plotWalk.py]: not Python3 compliant

File: doc/workshop/optimizer/exercise/plotWalk.py [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+trails = []
+clr = ax._get_lines.prop_cycler
+for case in cases:
+  c = next(clr)['color']
+  point, = ax.plot([data[case]['angle'][0]],[data[case]['v0'][0]],color='k',marker='${}$'.format(case))
+  trail, = ax.plot([data[case]['angle'][0]],[data[case]['v0'][0]],'-',color='k',alpha=0.5)
+  points.append(point)
+  trails.append(trail)
+
+def update_point(n,data,points,trails):
```
Comment (alfoa) [inline doc/workshop/optimizer/exercise/plotWalk.py]: camelBack

File: doc/workshop/reducedOrderModeling/inputs/examples/1_sample_Function.xml [hunk 20]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Simulation verbosity="debug">
+
+    <RunInfo>
+
+	    <WorkingDir>r1</WorkingDir>
```
Comment (alfoa) [inline doc/workshop/reducedOrderModeling/inputs/examples/1_sample_Function.xml]: indentation

Comment (alfoa) [reply doc/workshop/reducedOrderModeling/inputs/examples/1_sample_Function.xml]: <TestInfo>

File: doc/workshop/reducedOrderModeling/inputs/examples/2_train_rom.xml [hunk 21]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Simulation verbosity="debug">
+
+    <RunInfo>
+	    <WorkingDir>r2</WorkingDir>
```
Comment (alfoa) [inline doc/workshop/reducedOrderModeling/inputs/examples/2_train_rom.xml]: same as above

File: doc/workshop/reducedOrderModeling/inputs/examples/3_pickle_rom.xml [hunk 22]
```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Simulation verbosity="debug">
+
+    <RunInfo>
+	    <WorkingDir>r3</WorkingDir>
```
Comment (alfoa) [inline doc/workshop/reducedOrderModeling/inputs/examples/3_pickle_rom.xml]: samve as above

File: doc/workshop/reducedOrderModeling/inputs/examples/5_sample_rom.xml [hunk 23]
```diff
@@ -0,0 +1,131 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Simulation verbosity="debug">
+
+    <RunInfo>
+	    <WorkingDir>r5</WorkingDir>
+    	<Sequence>FirstMRun,trainRom,dumpRom,loadRom,sampleRom, plot</Sequence>
```
Comment (alfoa) [inline doc/workshop/reducedOrderModeling/inputs/examples/5_sample_rom.xml]: all the same as above

**Comment by alfoa** [comment]
I think the following files should be remove:

```
BIN doc/workshop/forwardSampling/mc_C.mp4 
Binary file not shown. 

BIN doc/workshop/forwardSampling/one_shot.mp4 
Binary file not shown. 
```

And for the presentations, I would keep the pptx files only and not also the PDFs. For example,
```
BIN doc/workshop/timeDepDataAnalysis/time_dep_relational_analysis.pdf 
Binary file not shown. 


BIN doc/workshop/timeDepDataMining/data_analysis.pdf 
Binary file not shown. 

```

File: doc/workshop/ExternalModels/projectile.py (hunk 1 above)
Comment (wangcj05) [reply doc/workshop/ExternalModels/projectile.py]: It can be merged, but I would like to keep a copy at this Folder just for the convenient.

**Comment by wangcj05** [comment]
Test info have been added to all tests. PDF files are removed, .mp4 files are also removed. The projectile model in the code coupling is merged with the projectile model in the ExternalModels. In addition, I did a clean up on the workshop materials.

**Comment by wangcj05** [comment]
All the XML input files in the doc/workshop/ are converted and added as regression tests except the following:
```
doc/workshop/timeDepDataMining/inputs/TD_KMeans.xml
```
This test is currently crashed, and a issue is reported in #799, a further PR will resolve this issue.

**Comment by wangcj05** [comment]
@alfoa Can you review it again?

**Comment by alfoa** [comment]
@wangcj05 some tests are failing

**Comment by wangcj05** [comment]
@alfoa The DTW test on mac is time out, I may need to move it to heavy test. There is another test failed on Fedora and I have no idea why: 
```
    DIFF framework.testExternalModel FAILED (
DIFF in /home/civet/civet/build_0/raven/tests/framework/externalModel/testPrintHistorySet_dump.csv: 
  Could not find match for row "101" in Gold:
x0                                            3.15838
y0                                            3.15838
z0                                            3.15838
ProbabilityWeight                            0.042875
filename             testPrintHistorySet_dump_100.csv
dtype: object) [FINISHED]
```

**Comment by alfoa** [comment]
@wangcj05 are we sure it is not another "not caught" failed test? can you try to compare the results manually?

**Comment by wangcj05** [comment]
@alfoa In my local mac machine, I have compared the results, and they look fine to me.

**Comment by alfoa** [comment]
----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. ***Ok***
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). ***n/a***
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. ***ok***
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. ***they pass***
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. ***n/a***
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. ***n/a***
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. ***#796***
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? ***n/a***

**Comment by alfoa** [comment]
Checklist passed...

Merging....

**Pull request merged**